tests/commonTemplates_test.go

```diff
@@ -299,9 +306,54 @@ var _ = Describe("Common templates", func() {
 
 				return false
 			}, shortTimeout).Should(BeTrue(), "ownerReference was not replaced by owner annotations on the old template")
+		})
+		It("should remove labels from old templates", func() {
+			triggerReconciliation()
+			// Template should eventually be updated by the operator
+			Eventually(func() bool {
+				updatedTpl := &templatev1.Template{}
+				key, err := client.ObjectKeyFromObject(oldTpl)
+				Expect(err).ToNot(HaveOccurred(), "failed to read template object key")
+				err = apiClient.Get(ctx, key, updatedTpl)
+				if err != nil {
+					return false
+				}
+				if updatedTpl.Labels["os.template.kubevirt.io/some-os"] == "" && updatedTpl.Labels["flavor.template.kubevirt.io/test"] == "" && updatedTpl.Labels["workload.template.kubevirt.io/server"] == "" && updatedTpl.Labels["template.kubevirt.io/type"] == "base" && updatedTpl.Labels["template.kubevirt.io/version"] == "not-latest" {
+					return true
+				}
+				return false
+			}, shortTimeout).Should(BeTrue(), "ownerReference was not replaced by owner annotations on the old template")
+		})
+		It("should continue to have labels on latest templates", func() {
+			baseRequirement, err := labels.NewRequirement("template.kubevirt.io/type", selection.Equals, []string{"base"})
+			Expect(err).To(BeNil())
+			var versionRequirement *labels.Requirement
+			versionRequirement, err = labels.NewRequirement("template.kubevirt.io/version", selection.Equals, []string{commonTemplates.Version})
+			Expect(err).To(BeNil())
+			labelsSelector := labels.NewSelector().Add(*baseRequirement, *versionRequirement)
+			opts := client.ListOptions{
+				LabelSelector: labelsSelector,
+			}
```

Comment:
To make the test more efficient, it could list the templates only in one namespace.
```
opts := client.ListOptions{
    LabelSelector: labelsSelector,
    Namespace: strategy.GetTemplatesNamespace(),
}
```